wilsonChan/assignments/reverseLinked/lc206/lc206.py

```diff
@@ -1,6 +1,49 @@
 # Problem Statement #
 
-# Given the head of a Singly LinkedList, reverse the LinkedList. Write a function to return the new head of the reversed LinkedList.
+# Given the head of a Singly LinkedList, reverse the LinkedList. 
+# Write a function to return the new head of the reversed LinkedList.
+
+'''
+Given:
+-Head of linked list
+
+Result:
+-Reverse the given linked list
+
+
+Example:
+Input: [1] [2] [3] [4] [5] [6]
+        H                   T
+
+Output: [6] [5] [4] [3] [2] [1]
+         H                   T
+
+
+Steps / Tasks:
+-Initialize the variables previous (None) and current (Head)
+-Create a while loop that will proceed when there are existing nodes
+  (Visualize zig zag pattern in code)
+  -Initialize the next variable to connect with the next node after the head
+  -Connected next node to variable previous (create link between variables next and previous)
+  -Connect variables previous to current
+    -this connection wasn't possible without linking variables next and previous.
+  -Connect the variables current to next
+
+
+
+Edge case:
+-If there is no value in head node or the node after it
+  -return head (1st node)
+
+
+Complexities:
+Time: O(n) - The size of the linked list can be any size
+
+Space: O(1) - The given linked list is a fixed size and doesn't change
+
+
```

Comment:
Great approach! :100: Love the walkthrough with an example, the edge case, and the complexities written out.